### DISCUSSION TITLE
Number of consumers

### DISCUSSION THREAD

**Discussion by niondir** [nats-io/nats-server#6240] [Q&A]
Hi,
I'm a bit concerned about a current setup we have, since it is based on a lot of consumers.

We have Organisations that have one subject each like `integration.{orgId}` and we are creating one consumer per organisation, bound to a single subject each.
This can easily be a few hundred. 

Does this make sense at all? Does it scale well? Or should we consider to work with less consumers and have more like one consumer per CPU core.

The reason behind this discission was, that when one customer (Consumer) is blocking or taking long time for processing messages, the others can still operate.

I'm happy for any feedback about this setup.

**Comment by derekcollison** [comment]
What is the retention policy of the stream?
What is the replica count of the stream?
How many orgs in total do you expect?

**Comment by niondir** [comment]
The stream is a Working Queue without replica yet.

I expect something between 100-1.000 Organisations. Should stay below 10k

**Reply by derekcollison** [reply]
That should be fine..
